This upstream patch addresses Launchpad bug 1244304. Although it's been
addressed in Icehouse 2014.1, the patch below is still not yet released
for Havana.

commit 380894fa4a9329c963fad7dbd17fd9b94b1918a8
Author: Kun Huang <gareth@unitedstack.com>
Date:   Fri Oct 25 00:26:43 2013 +0800

    remove 8888 port in sample_data.sh
    
    Swift now doesn't use 8888 port as any default service. And the origin
    sample_data.sh always let new users to fail and modify swift bind port.
    So switching 8888 port here to 8080 could generate sample data without
    any change for building swift + keystone environment.
    
    Closes-Bug: #1244304
    Change-Id: If58f9f025f57565733aa25efc2fdf06865781391

diff --git a/keystone/tests/test_token_provider.py b/keystone/tests/test_token_provider.py
index c93bd73..8b2c212 100644
--- a/keystone/tests/test_token_provider.py
+++ b/keystone/tests/test_token_provider.py
@@ -92,10 +92,10 @@ SAMPLE_V2_TOKEN = {
             {
                 "endpoints": [
                     {
-                        "adminURL": "http://localhost:8888/v1",
+                        "adminURL": "http://localhost:8080/v1",
                         "id": "7bd0c643e05a4a2ab40902b2fa0dd4e6",
-                        "internalURL": "http://localhost:8888/v1/AUTH_01257",
-                        "publicURL": "http://localhost:8888/v1/AUTH_01257",
+                        "internalURL": "http://localhost:8080/v1/AUTH_01257",
+                        "publicURL": "http://localhost:8080/v1/AUTH_01257",
                         "region": "RegionOne"
                     }
                 ],
@@ -202,19 +202,19 @@ SAMPLE_V3_TOKEN = {
                         "id": "7bd0c643e05a4a2ab40902b2fa0dd4e6",
                         "interface": "admin",
                         "region": "RegionOne",
-                        "url": "http://localhost:8888/v1"
+                        "url": "http://localhost:8080/v1"
                     },
                     {
                         "id": "43bef154594d4ccb8e49014d20624e1d",
                         "interface": "internal",
                         "region": "RegionOne",
-                        "url": "http://localhost:8888/v1/AUTH_01257"
+                        "url": "http://localhost:8080/v1/AUTH_01257"
                     },
                     {
                         "id": "e63b5f5d7aa3493690189d0ff843b9b3",
                         "interface": "public",
                         "region": "RegionOne",
-                        "url": "http://localhost:8888/v1/AUTH_01257"
+                        "url": "http://localhost:8080/v1/AUTH_01257"
                     }
                 ],
                 "id": "a669e152f1104810a4b6701aade721bb",
@@ -399,10 +399,10 @@ SAMPLE_V2_TOKEN_WITH_EMBEDED_VERSION = {
             {
                 "endpoints": [
                     {
-                        "adminURL": "http://localhost:8888/v1",
+                        "adminURL": "http://localhost:8080/v1",
                         "id": "7bd0c643e05a4a2ab40902b2fa0dd4e6",
-                        "internalURL": "http://localhost:8888/v1/AUTH_01257",
-                        "publicURL": "http://localhost:8888/v1/AUTH_01257",
+                        "internalURL": "http://localhost:8080/v1/AUTH_01257",
+                        "publicURL": "http://localhost:8080/v1/AUTH_01257",
                         "region": "RegionOne"
                     }
                 ],
@@ -509,19 +509,19 @@ SAMPLE_V3_TOKEN_WITH_EMBEDED_VERSION = {
                         "id": "7bd0c643e05a4a2ab40902b2fa0dd4e6",
                         "interface": "admin",
                         "region": "RegionOne",
-                        "url": "http://localhost:8888/v1"
+                        "url": "http://localhost:8080/v1"
                     },
                     {
                         "id": "43bef154594d4ccb8e49014d20624e1d",
                         "interface": "internal",
                         "region": "RegionOne",
-                        "url": "http://localhost:8888/v1/AUTH_01257"
+                        "url": "http://localhost:8080/v1/AUTH_01257"
                     },
                     {
                         "id": "e63b5f5d7aa3493690189d0ff843b9b3",
                         "interface": "public",
                         "region": "RegionOne",
-                        "url": "http://localhost:8888/v1/AUTH_01257"
+                        "url": "http://localhost:8080/v1/AUTH_01257"
                     }
                 ],
                 "id": "a669e152f1104810a4b6701aade721bb",
diff --git a/tools/sample_data.sh b/tools/sample_data.sh
index d09502d..65030b5 100755
--- a/tools/sample_data.sh
+++ b/tools/sample_data.sh
@@ -215,9 +215,9 @@ keystone service-create --name=swift \
                         --description="Swift Service")
 if [[ -z "$DISABLE_ENDPOINTS" ]]; then
     keystone endpoint-create --region RegionOne --service-id $SWIFT_SERVICE \
-        --publicurl   "http://$CONTROLLER_PUBLIC_ADDRESS:8888/v1/AUTH_\$(tenant_id)s" \
-        --adminurl    "http://$CONTROLLER_ADMIN_ADDRESS:8888/v1" \
-        --internalurl "http://$CONTROLLER_INTERNAL_ADDRESS:8888/v1/AUTH_\$(tenant_id)s"
+        --publicurl   "http://$CONTROLLER_PUBLIC_ADDRESS:8080/v1/AUTH_\$(tenant_id)s" \
+        --adminurl    "http://$CONTROLLER_ADMIN_ADDRESS:8080/v1" \
+        --internalurl "http://$CONTROLLER_INTERNAL_ADDRESS:8080/v1/AUTH_\$(tenant_id)s"
 fi
 
 # create ec2 creds and parse the secret and access key returned
